app/controllers/account_activations_controller.rb

```diff
@@ -1,16 +1,22 @@
 class AccountActivationsController < ApplicationController
-
+  before_action :active_account, only: %i[edit]
   def edit
     user = User.find_by(email: params[:email])
     if user && !user.activated? && user.authenticated?(:activation, params[:id])
-      user.update_attribute(:activated,    true)
-      user.update_attribute(:activated_at, Time.zone.now)
-      log_in user
-      flash[:success] = t ".account_activated"
-      redirect_to login_path
+      active_account
     else
-      flash[:danger] = t ".invalid_activation_link"
+      flash[:danger] = t '.invalid_activation_link'
       redirect_to login_path
     end
   end
+
+  private
+
+  def active_account
```

Comment:
- recommend rename to: `user_active_account` for clear
- the method needs a parameter
- it's better to place the method in `user.rb` model